core/model/modx/processors/resource/delete.class.php

```diff
@@ -256,12 +256,15 @@ public function fireAfterDelete(array $childrenIds = array()) {
      * @return void
      */
     public function clearCache() {
-        $this->modx->cacheManager->refresh(array(
-            'db' => array(),
-            'auto_publish' => array('contexts' => array($this->resource->get('context_key'))),
-            'context_settings' => array('contexts' => array($this->resource->get('context_key'))),
-            'resource' => array('contexts' => array($this->resource->get('context_key'))),
-        ));
+	$clear = $this->getProperty('syncsite',$this->modx->getOption('syncsite_default')) || $this->getProperty('clearCache',false);
+        if ($clear) {
+            $this->modx->cacheManager->refresh(array(
+                'db' => array(),
+                'auto_publish' => array('contexts' => array($this->workingContext->get('key'))),
+                'context_settings' => array('contexts' => array($this->workingContext->get('key'))),
+                'resource' => array('contexts' => array($this->workingContext->get('key'))),
```

Comment:
While in the process of porting this to 3.x, I noticed these lines were changed from the resource context_key to the working context. I think this was a mistake and an oversight. Does anyone else agree?

Reply:
Oh, good catch. I definitely overlooked that, and don't see any definition of $this->workingContext anywhere either?

Reply:
Yeah. I'll fix that up and just push it directly.